eval/eval.go

```diff
@@ -352,6 +362,32 @@ func evalArrayIndexExpression(array, index object.Object) object.Object {
 	return arrayObject.Elements[idx]
 }
 
+func (s *State) applyExtension(fn object.Extension, args []object.Object) object.Object {
+	l := len(args)
+	if l < fn.MinArgs {
+		return object.Error{Value: fmt.Sprintf("<wrong number of arguments got=%d, want %s>",
+			l, fn.Inspect())} // shows usage
+	}
+	if fn.MaxArgs != -1 && l > fn.MaxArgs {
+		return object.Error{Value: fmt.Sprintf("<wrong number of arguments got=%d, want %s>",
+			l, fn.Inspect())} // shows usage
+	}
+	for i, arg := range args {
+		if i >= len(fn.ArgTypes) {
+			break
+		}
+		if fn.ArgTypes[i] == object.FLOAT && arg.Type() == object.INTEGER {
+			args[i] = object.Float{Value: float64(arg.(object.Integer).Value)}
```

Comment:
Please add a comment explaining why you are converting to float64 here

Reply:
done, thx